cache.h

```diff
@@ -311,6 +311,28 @@ struct untracked_cache;
 struct progress;
 struct pattern_list;
 
+enum sparse_index_mode {
+	/*
+	 * COMPLETELY_FULL: there are no sparse directories
+	 * in the index at all.
+	 */
+	COMPLETELY_FULL = 0,
+
+	/*
+	 * COMPLETELY_SPARSE: the sparse directories that exist
+	 * exactly match the sparse-checkout boundary.
+	 */
+	COMPLETELY_SPARSE = 1,
```

Comment:
I think there's a distinction between "completely collapsed" (as in, files outside the sparse checkout boundary might be in the index, but it's because they have modifications/are otherwise uncollapsible) and "completely sparse" (_nothing_ exists outside of the sparse checkout boundary). As far as I can tell, this setting refers to a state of "completely collapsed" (especially because `PARTIALLY_SPARSE` is described with "it is possible that some file entries could collapse to sparse directory entries."). The name `COMPLETELY_SPARSE` is probably fine as-is, but if it is really "completely collapsed", can you clarify that in the description?

Reply:
Names are hard. I like your suggestion.